crypto/fipsmodule/rand/rand.c

```diff
@@ -395,7 +403,10 @@ void RAND_bytes_with_additional_data(uint8_t *out, size_t out_len,
     // entropy is used. This can be expensive (one read per |RAND_bytes| call)
     // and so is disabled when we have fork detection, or if the application has
     // promised not to fork.
-    if (fork_generation != 0 || fork_unsafe_buffering) {
+    // snapsafe_status is only 0 when the kernel has snapsafe support, but it
+    // failed to initialize. Otherwise, snapsafe_status is 1.
+    if ((snapsafe_status != 0 && fork_generation != 0) ||
+        fork_unsafe_buffering) {
       OPENSSL_memset(additional_data, 0, sizeof(additional_data));
```

Comment:
I realize you didn't add this but you are just continuing the pattern: why the memset 0? Even if we have fork protection or snapsafe why not just mix in whatever uninitialized value was in memory?

Reply:
The memset to 0 is done to avoid collecting extra data/entropy via a slow syscall/rdrand invocation unless it's required. So this change essentially makes it required when snapsafe is supported but not initialized properly.

Reply:
Well that's just what this branch does, you could avoid collecting slow system entropy *and* avoid calling memset. Nothing I see in the rest of this function depends on this additional_data being all 0 if there is no system entropy to collect.